### PR TITLE
First draft/skeleton of a contributor guide

### PR DESCRIPTION
**Pull request by ceedubs**
It still has a lot of TODOs, but I'm a fan of getting something out
there and letting people iterate on it. None of this is set in stone -
I'm fully expecting that even out of the small amount that's here people
will suggest improvements.

### CODE REVIEW

**Comment by non** [comment]
One small piece of feedback: I feel like the introduction here almost works better as a conclusion.

I would rather jump right in, as opposed to starting off talking about the reasons contributing to an open source project is hard.

I think your prose is good, I just think it works better as a conclusion, e.g. "yes, this process can be hard, but if we've done our job, this document has clearly laid it all out, and you shouldn't have any problems."

**Comment by ceedubs** [comment]
@non thanks for the good feedback. I can hopefully incorporate it into a commit this evening (UTC-05:00)

**Comment by mandubian** [comment]
@ceedubs Let's add that someone submitting a PR solving an issue must associate it to the issue by adding a commit or a PR description containing "fixes #53" for example

**Comment by mandubian** [comment]
@ceedubs Let's add that people taking in charge a PR should at least put a comment in the PR saying it

**Comment by non** [comment]
Yeah both of @mandubian's comments are on point. Maybe we should even put something in the `README` about it too, since it mentions getting involved?

**Comment by mandubian** [comment]
+1 for the README

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+
+## process
+
+Cats follows a pretty standard [fork and pull](https://help.github.com/articles/using-pull-requests/) model for contributions via GitHub pull requests.
+
+Below is a list of the steps that might be involved in an ideal contribution. If you don't have the time to go through every step, feel free to contribute what you can, and someone else will probably be happy to follow up with any polishing that may need to be done. If you want to touch up some documentation or fix typos, please feel free to skip all of these steps and jump straight to subitting a pull request.
```
Comment (travisbrown) [inline CONTRIBUTING.md]: `s/subitting/submitting/`

**Comment by travisbrown** [comment]
As a fellow member of the robotic consistency school I'd prefer to see standardized capitalization for headers. I'd prefer first word only, but the README does both first word and every word.

**Comment by travisbrown** [comment]
One other tiny thing: enforcing a max line length here would make diffs and line-level comments a little better.

**Comment by tpolecat** [comment]
Love it. Every project needs a consistent robot posse.

**Comment by travisbrown** [comment]
@tpolecat It's better if it's made up of, you know, actual robots.

**Comment by tpolecat** [comment]
I for one welcome our consistent overlords.

**Comment by ceedubs** [comment]
I've updated the PR based on feedback.

There are still several sections that are just TODOs.

Also I didn't break paragraphs based on line length as was suggested. To me it seems like a bit of a hassle for prose (as opposed to code). It can certainly be changed if that's what people prefer, though.

Considering that we've had already had some issues with duplicate effort, I'd be a little inclined to get this out there and let people fill in the missing pieces. The TODOs are kind of an eyesore, so I understand if others don't like that approach. I could fill out the "Write code" section a bit, but it seems like we are still trying to settle on how we are approaching the other TODO items.

**Comment by non** [comment]
No I think you're right. Something is better than nothing at this stage. Let's merge it and iterate! :+1:

**Comment by stew** [comment]
:+1:

**Pull request merged**